### ISSUE TITLE
os.Executable is currently broken in go, even with 1.8

### ISSUE THREAD

**Issue by darinkes** [kardianos/osext#22]
The Commit https://github.com/kardianos/osext/commit/9b883c5eb462dd5cb1b0a7a104fe86bc6b9bd391 breaks the fix for OpenBSD in https://github.com/kardianos/osext/commit/b4814f465fb1f92d46e37f7ef84d732ece7c3e3a  since go still expects procfs on OpenBSD (see: https://github.com/golang/go/blob/master/src/os/executable_procfs.go#L26)

I already opened an Issue (https://github.com/golang/go/issues/19453) but still would suggest to use the osext.Executable for go1.8+ on OpenBSD.
```
diff --git a/osext_go18.go b/osext_go18.go
index 7065a5f..cb1d168 100644
--- a/osext_go18.go
+++ b/osext_go18.go
@@ -1,3 +1,4 @@
+//+build !openbsd
 //+build go1.8

 package osext
diff --git a/osext_sysctl.go b/osext_sysctl.go
index 579f224..33cee25 100644
--- a/osext_sysctl.go
+++ b/osext_sysctl.go
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.

-// +build !go1.8,darwin !go1.8,freebsd !go1.8,openbsd
+// +build !go1.8,darwin !go1.8,freebsd openbsd

 package osext
```

go test without the fix:
```
--- FAIL: TestPrintExecutable (0.00s)
        osext_test.go:30: Executable failed: readlink /proc/curproc/file: no such file or directory
--- FAIL: TestPrintExecutableFolder (0.00s)
        osext_test.go:37: ExecutableFolder failed: readlink /proc/curproc/file: no such file or directory
--- FAIL: TestExecutableFolder (0.00s)
        osext_test.go:44: ExecutableFolder failed: readlink /proc/curproc/file: no such file or directory
--- FAIL: TestExecutableMatch (0.00s)
        osext_test.go:53: Executable failed: readlink /proc/curproc/file: no such file or directory
FAIL
exit status 1
FAIL    github.com/kardianos/osext      0.017s
```

**Comment by kardianos** [comment]
Please try the `kardianos-openbsd` branch and verify it is working for you.

**Comment by darinkes** [comment]
Thanks! It works

```
|-- srinkes@openbsd-amd64-dev:~/osext (kardianos-openbsd) --|$ go test
PASS
ok      _/home/srinkes/osext    0.078s
|-- srinkes@openbsd-amd64-dev:~/osext (kardianos-openbsd) --|$ go version
go version go1.8 openbsd/amd64
|-- srinkes@openbsd-amd64-dev:~/osext (kardianos-openbsd) --|$
```

**Comment by xenoscopic** [comment]
It looks like the Go team's fix ([https://go-review.googlesource.com/#/c/46004/](https://go-review.googlesource.com/#/c/46004/)) is going to land in Go 1.9.  At first glance it seems like the same approach as that in `osext`, so perhaps it's worth re-enabling the `os.Executable` implementation for OpenBSD in 1.9 (the current fix from #23 will continue to use the `osext` implementation in post-1.8 versions).
